src/libcore/slice/mod.rs

```diff
@@ -2652,6 +2666,199 @@ impl<T> [T] {
     {
         self.iter().is_sorted_by_key(f)
     }
+
+    /// Removes the portion of the slice specified by the range provided.
+    ///
+    /// Returns the portion of the slice removed, or `None` if the provided
+    /// bound starts or ends outside the slice's length.
```

Comment:
This should mention that the original slice is not modified if `None` is returned. Maybe something like this? Just a suggestion:

> Removes and returns the portion of the slice specified by the given range.
>
> If the `range` starts or ends outside of this slice, `None` is returned and this slice is not modified. Otherwise, `Some(self[range])` is returned and this slice is modified to only contain the elements not included in `range`.